target/cheri-common/op_helper_cheri_common.c

```diff
@@ -1271,5 +1271,43 @@ void store_cap_to_memory(CPUArchState *env, uint32_t cs,
 }
 #endif
 
+void CHERI_HELPER_IMPL(raise_exception_pcc_perms(CPUArchState *env))
+{
+    // On translation block entry we check that PCC is tagged and unsealed,
+    // has the required permissions and is within bounds
+    // The running of the end check is performed in the translator
+    cheri_debug_assert(pc_is_current(env));
+    const cap_register_t *pcc = cheri_get_current_pcc(env);
+    CheriCapExcCause cause;
+    if (!pcc->cr_tag) {
+        cause = CapEx_TagViolation;
+    } else if (!cap_is_unsealed(pcc)) {
+        cause = CapEx_SealViolation;
```

Comment:
Sentries are unsealed by jumps?

Reply:
Yes, helper_cjalr/cheri_update_pcc_for_exc_return should have unsealed them before moving the value into PCC.